tests/unit/processors/test_files_processor.py

```diff
@@ -0,0 +1,115 @@
+from gitlabform.processors.project.files_processor import FilesProcessor
+from gitlabform.gitlab.core import NotFoundException
+
+class MockGitLab:
+    """
+    A mock GitLab class to simulate the behavior of GitLab API for unit testing.
+    """
+    def __init__(self, branches):
+        self.branches = branches
+    
+    def get_branch(self, project_and_group, branch):
+        # Simulate the response for getting a branch
+        if branch in self.branches:
+            return self.branches[branch]
+        else:
+            raise NotFoundException(f"Branch '{branch}' not found.")
+
+class FilesProcessor:
+    def __init__(self):
+        # Create an instance of FilesProcessor with a mock GitLab instance
+        self.mock_gitlab = MockGitLab(
+            branches={
+                "main": {
+                    "protected": True,
+                    "push_access_levels": [{"access_level_description": "no one"}],
+                    "merge_access_levels": [{"access_level_description": "maintainers"}]
+                },
+                "dev": {
+                    "protected": False,
+                    "push_access_levels": [{"access_level_description": "developers"}],
+                    "merge_access_levels": [{"access_level_description": "maintainers"}]
+                },
+                "feature": {
+                    "protected": True,
+                    "push_access_levels": [{"access_level_description": "developers"}],
+                    "merge_access_levels": [{"access_level_description": "developers"}]
+                },
+            }
+        )
+        self.processor = FilesProcessor(self.mock_gitlab, None, True) # Config and strict mode don't affect pre_protection_check
+    
+    def run_tests(self):
```

Comment:
probably not required, pytest _should_ execute everything in this file already right?